intersection_transit_maneuvering/src/intersection_transit_maneuvering.cpp

```diff
@@ -0,0 +1,214 @@
+    {
+        
+        ros::WallTime start_time = ros::WallTime::now();  // Start timeing the execution time for planning so it can be logged
+
+        lanelet::BasicPoint2d veh_pos(req.vehicle_state.X_pos_global, req.vehicle_state.Y_pos_global);
+        double current_downtrack = wm_->routeTrackPos(veh_pos).downtrack;
+
+        std::vector<cav_msgs::Maneuver> maneuver_plan;
+        for(size_t i = req.maneuver_index_to_plan; i < req.maneuver_plan.maneuvers.size(); i++)
+        {
+            if(req.maneuver_plan.maneuvers[i].type == cav_msgs::Maneuver::INTERSECTION_TRANSIT_STRAIGHT ||
+            req.maneuver_plan.maneuvers[i].type == cav_msgs::Maneuver::INTERSECTION_TRANSIT_LEFT || 
+            req.maneuver_plan.maneuvers[i].type == cav_msgs::Maneuver::INTERSECTION_TRANSIT_RIGHT )
+            {
+                maneuver_plan.push_back(req.maneuver_plan.maneuvers[i]);
+                resp.related_maneuvers.push_back(i);
+            }
+            else
+                {
+                    break;
+                }
+        }
+        cav_srvs::PlanTrajectoryRequest& req2;
+        auto new_plan = convert_maneuver_plan(maneuver_plan);
+
+        for(const auto& man : new_plan)
+            {
+                req2.maneuver_plan.push_back(man);
+            }
+
+        /*Once the maneuvers have been successfully converted, call the inlanecruising plugin to calculate the trajectory*/
+        inlanecruising_plugin::InLaneCruisingPlugin::plan_trajectory_cb(req2, resp);
+        
+        return true;
+    }
+    
+
+    void IntersectionTransitManeuvering::run()
+    {
+        /**/
+        initialize();
+        ros::CARMANodeHandle::spin();
+    
+    }
+
+   
+    std::vector<cav_msgs::Maneuver> IntersectionTransitManeuvering::convert_maneuver_plan(const std::vector<cav_msgs::Maneuver>& maneuvers)
+    {
+        if (maneuvers.size().isEmpty())
+        {
+            throw std::invalid_argument("No maneuvers to convert");
+        }
+
+        std::vector<cav_msgs::LaneFollowingManeuver> new_maneuver_plan;
+        cav_msgs:LaneFollowingManeuver new_maneuver;
+        for(const auto& maneuver : maneuvers)
+        {
+            /*Throw exception if the manuever type does not match INTERSECTION_TRANSIT*/
+            if ( maneuver.type != cav_msgs::Maneuver::INTERSECTION_TRANSIT_STRAIGHT ||
+                maneuver.type != cav_msgs::Maneuver::INTERSECTION_TRANSIT_LEFT_TURN ||
+                maneuver.type != cav_msgs::Maneuver::INTERSECTION_TRANSIT_RIGHT_TURN ||)
+                {
+                    throw std::invalid_argument("Intersection transit maneuvering does not support this maneuver type");
+                }
+            
+            /*Convert IT Straight*/
+            if (maneuver.type == cav_msgs::Maneuver::INTERSECTION_TRANSIT_STRAIGHT)
+            {
+                new_maneuver.parameters = maneuver.intersection_transit_straight_maneuver.parameters;
```

Comment:
probably best to clear the presence_vector and meta data fields as these may not be applicable to inlane-cruising.

I'm going back and forth on whether we should set a new maneuver id as well, it would help for the id to be persistent to support logging, but could potentially lead to issues if misinterpreted. Check with @kjrush if the forwarded lane follow maneuver should have its own maneuver id.